### PR TITLE
Fixes for XA transactions with streaming enabled

### PR DESCRIPTION
**Pull request by pacheco**
Changes mostly related to handling of XA PREPARE fragments

### CODE REVIEW

File: dbsim/db_client_service.hpp [hunk 1]
```diff
@@ -46,6 +46,10 @@ namespace db
         {
             return false;
         }
+        bool is_xa_prepare () const override
```
Comment (sciascid) [inline dbsim/db_client_service.hpp]: extra space: "is_xa_prepare ()" -> "is_xa_prepare()"

File: src/transaction.cpp [hunk 2]
```diff
@@ -182,6 +182,16 @@ int wsrep::transaction::start_transaction(
...
+int wsrep::transaction::next_fragment(
+    const wsrep::ws_meta& ws_meta)
+{
+    debug_log_state("next_fragment enter");
+    ws_meta_ = ws_meta;
+    debug_log_state("next_fragment leave");
+    return 0;
+}
+
+
```
Comment (sciascid) [inline src/transaction.cpp]: extra new line

File: src/transaction.cpp [hunk 3]
```diff
@@ -1086,12 +1096,12 @@ int wsrep::transaction::streaming_step(wsrep::unique_lock<wsrep::mutex>& lock)
...
     }
 
-    if (streaming_context_.fragment_size_exceeded())
+    if (streaming_context_.fragment_size_exceeded() || client_service_.is_xa_prepare())
     {
         // Some statements have no effect. Do not atttempt to
         // replicate a fragment if no data has been generated
-        // since last fragment replication.
-        if (bytes_to_replicate <= 0)
+        // since last fragment replication, and statement is not XA PREPARE.
```
Comment (sciascid) [inline src/transaction.cpp]: This comment is not very useful. Briefly explain that XA PREPARE may add some data down this path

File: test/mock_high_priority_service.cpp [hunk 4]
```diff
@@ -26,6 +26,13 @@ int wsrep::mock_high_priority_service::start_transaction(
     return client_state_->start_transaction(ws_handle, ws_meta);
 }
 
+int wsrep::mock_high_priority_service::next_fragment(const wsrep::ws_meta& ws_meta)
+{
+    return client_state_->next_fragment(ws_meta);
+}
+
+
```
Comment (sciascid) [inline test/mock_high_priority_service.cpp]: new lines

**Comment by pacheco** [comment]
i've commented out the current thread assertion in next_fragment as we discussed.

**Pull request merged**